From d063bf429a8d1cc2b651a2ffa542cc98bd369613 Mon Sep 17 00:00:00 2001
From: Francois Choleau <francois.choleau@st.com>
Date: Fri, 21 Jun 2024 05:33:12 -0400
Subject: [PATCH] arm64: dts: st: update stm32mp23 SoCs family

alignement vs stm32mp25 soc dts

Signed-off-by: Francois Choleau <francois.choleau@st.com>
Change-Id: I0859c3416edf9a2cdcaf56330834eece6b767ac4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/388963
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp231.dtsi | 191 +++++++++++++++++--------
 arch/arm64/boot/dts/st/stm32mp233.dtsi |   7 +-
 2 files changed, 135 insertions(+), 63 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp231.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp231.dtsi
@@ -36,10 +36,9 @@
 				compatible = "arm,idle-state";
 				arm,psci-suspend-param = <0x00000001>;
 				local-timer-stop;
-				/* TODO: set correct TIMING */
-				entry-latency-us = <10>;
-				exit-latency-us = <10>;
-				min-residency-us = <100>;
+				entry-latency-us = <300>;
+				exit-latency-us = <500>;
+				min-residency-us = <1000>;
 			};
 		};
 
@@ -47,28 +46,25 @@
 			STOP1: domain-stop1 {
 				compatible = "domain-idle-state";
 				arm,psci-suspend-param = <0x00000011>;
-				/* TODO: set correct TIMING */
-				entry-latency-us = <20>;
-				exit-latency-us = <20>;
-				min-residency-us = <60>;
+				entry-latency-us = <400>;
+				exit-latency-us = <1200>;
+				min-residency-us = <1500>;
 			};
 
 			LP_STOP1: domain-lp-stop1 {
 				compatible = "domain-idle-state";
 				arm,psci-suspend-param = <0x0000021>;
-				/* TODO: set correct TIMING */
-				entry-latency-us = <1000>;
-				exit-latency-us = <1000>;
+				entry-latency-us = <500>;
+				exit-latency-us = <2000>;
 				min-residency-us = <3000>;
 			};
 
 			LPLV_STOP1: domain-lplv-stop1 {
 				compatible = "domain-idle-state";
 				arm,psci-suspend-param = <0x00000211>;
-				/* TODO: set correct TIMING */
-				entry-latency-us = <2000>;
-				exit-latency-us = <2000>;
-				min-residency-us = <6000>;
+				entry-latency-us = <500>;
+				exit-latency-us = <3000>;
+				min-residency-us = <4000>;
 			};
 		};
 	};
@@ -87,12 +83,6 @@
 	};
 
 	clocks {
-		clk_dsi_txbyte: txbyteclk {
-			#clock-cells = <0>;
-			compatible = "fixed-clock";
-			clock-frequency = <0>;
-		};
-
 		clk_rcbsec: clk-rcbsec {
 			#clock-cells = <0>;
 			compatible = "fixed-clock";
@@ -116,6 +106,11 @@
 			#size-cells = <0>;
 			linaro,optee-channel-id = <0>;
 
+			scmi_devpd: protocol@11 {
+				reg = <0x11>;
+				#power-domain-cells = <1>;
+			};
+
 			scmi_perf: protocol@13 {
 				reg = <0x13>;
 				#clock-cells = <1>;
@@ -158,10 +153,6 @@
 						reg = <VOLTD_SCMI_UCPD>;
 						regulator-name = "vdd33ucpd";
 					};
-					scmi_vdd33usb: regulator@6 {
-						reg = <VOLTD_SCMI_USB33>;
-						regulator-name = "vdd33usb";
-					};
 					scmi_vdda18adc: regulator@7 {
 						reg = <VOLTD_SCMI_ADC>;
 						regulator-name = "vdda18adc";
@@ -175,10 +166,12 @@
 		compatible = "arm,cortex-a7-gic";
 		#interrupt-cells = <3>;
 		interrupt-controller;
+		interrupt-parent = <&intc>;
 		reg = <0x0 0x4ac10000 0x0 0x1000>,
 		      <0x0 0x4ac20000 0x0 0x2000>,
 		      <0x0 0x4ac40000 0x0 0x2000>,
 		      <0x0 0x4ac60000 0x0 0x2000>;
+		interrupts = <GIC_PPI 9 (GIC_CPU_MASK_SIMPLE(1) | IRQ_TYPE_LEVEL_LOW)>;
 
 		#address-cells = <2>;
 		#size-cells = <2>;
@@ -269,20 +262,20 @@
 	usb2_phy1: usb2-phy1 {
 		compatible      = "st,stm32mp25-usb2phy";
 		#phy-cells      = <0>;
+		#clock-cells    = <0>;
 		st,syscfg       = <&syscfg 0x2400>;
 		clocks          = <&rcc CK_KER_USB2PHY1>;
 		resets          = <&rcc USB2PHY1_R>;
-		vdd33-supply    = <&scmi_vdd33usb>;
 		status          = "disabled";
 	};
 
 	usb2_phy2: usb2-phy2 {
 		compatible      = "st,stm32mp25-usb2phy";
 		#phy-cells      = <0>;
+		#clock-cells    = <0>;
 		st,syscfg       = <&syscfg 0x2800>;
 		clocks          = <&rcc CK_KER_USB2PHY2EN>;
 		resets          = <&rcc USB2PHY2_R>;
-		vdd33-supply    = <&scmi_vdd33usb>;
 		status          = "disabled";
 	};
 
@@ -1291,7 +1284,6 @@
 				interrupts = <GIC_SPI 157 IRQ_TYPE_LEVEL_HIGH>;
 				resets = <&rcc SAI1_R>;
 				access-controllers = <&rifsc 49>;
-				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
 				sai1a: audio-controller@40290004 {
@@ -1328,7 +1320,6 @@
 				interrupts = <GIC_SPI 162 IRQ_TYPE_LEVEL_HIGH>;
 				resets = <&rcc SAI2_R>;
 				access-controllers = <&rifsc 50>;
-				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
 				sai2a: audio-controller@402a0004 {
@@ -1365,7 +1356,6 @@
 				interrupts = <GIC_SPI 192 IRQ_TYPE_LEVEL_HIGH>;
 				resets = <&rcc SAI3_R>;
 				access-controllers = <&rifsc 51>;
-				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
 				sai3a: audio-controller@402b0004 {
@@ -1415,7 +1405,6 @@
 				interrupts = <GIC_SPI 223 IRQ_TYPE_LEVEL_HIGH>;
 				resets = <&rcc SAI4_R>;
 				access-controllers = <&rifsc 52>;
-				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 
 				sai4a: audio-controller@40340004 {
@@ -1502,6 +1491,8 @@
 					#io-channel-cells = <1>;
 					#address-cells = <1>;
 					#size-cells = <0>;
+					nvmem-cells = <&vrefint>;
+					nvmem-cell-names = "vrefint";
 					status = "disabled";
 					channel@14 {
 						reg = <14>;
@@ -1519,6 +1510,8 @@
 					#io-channel-cells = <1>;
 					#address-cells = <1>;
 					#size-cells = <0>;
+					nvmem-cells = <&vrefint>;
+					nvmem-cell-names = "vrefint";
 					status = "disabled";
 					channel@14 {
 						reg = <14>;
@@ -1643,6 +1636,8 @@
 					#io-channel-cells = <1>;
 					#address-cells = <1>;
 					#size-cells = <0>;
+					nvmem-cells = <&vrefint>;
+					nvmem-cell-names = "vrefint";
 					status = "disabled";
 					channel@14 {
 						reg = <14>;
@@ -1687,6 +1682,26 @@
 				status = "disabled";
 			};
 
+			iwdg1: watchdog@44010000 {
+				compatible = "st,stm32mp1-iwdg";
+				reg = <0x44010000 0x400>;
+				interrupts = <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_BUS_IWDG1>, <&scmi_clk CK_SCMI_LSI>;
+				clock-names = "pclk", "lsi";
+				access-controllers = <&rifsc 98>;
+				status = "disabled";
+			};
+
+			iwdg2: watchdog@44020000 {
+				compatible = "st,stm32mp1-iwdg";
+				reg = <0x44020000 0x400>;
+				interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_BUS_IWDG2>, <&scmi_clk CK_SCMI_LSI>;
+				clock-names = "pclk", "lsi";
+				access-controllers = <&rifsc 99>;
+				status = "disabled";
+			};
+
 			spi8: spi@46020000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -1974,37 +1989,81 @@
 				};
 			};
 
-			usb2h: usb@482e0000 {
-				compatible = "st,stm32mp25-usb2h";
+			usbh: usb@482e0000 {
+				compatible = "st,stm32mp25-usbh";
 				st,syscfg = <&syscfg 0x2420>;
 				#address-cells = <1>;
 				#size-cells = <1>;
 				ranges = <0x482e0000 0x482e0000 0x20000>;
 				access-controllers = <&rifsc 63>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
+				interrupts-extended = <&exti1 43 IRQ_TYPE_EDGE_RISING>;
 				status = "disabled";
 
-				usb2h_ohci: usb@482e0000 {
+				usbh_ohci: usb@482e0000 {
 					compatible = "generic-ohci";
 					reg = <0x482e0000 0x1000>;
-					clocks = <&rcc CK_BUS_USB2OHCI>;
+					clocks = <&usb2_phy1>, <&rcc CK_BUS_USB2OHCI>;
 					resets = <&rcc USB2_R>;
 					interrupts = <GIC_SPI 140 IRQ_TYPE_LEVEL_HIGH>;
 					phys = <&usb2_phy1>;
 					phy-names = "usb";
+					wakeup-source;
 				};
 
-				usb2h_ehci: usb@482f0000 {
+				usbh_ehci: usb@482f0000 {
 					compatible = "generic-ehci";
 					reg = <0x482f0000 0x1000>;
 					clocks = <&rcc CK_BUS_USB2EHCI>;
 					resets = <&rcc USB2_R>;
 					interrupts = <GIC_SPI 139 IRQ_TYPE_LEVEL_HIGH>;
-					companion = <&usb2h_ohci>;
+					companion = <&usbh_ohci>;
 					phys = <&usb2_phy1>;
 					phy-names = "usb";
+					wakeup-source;
 				};
 			};
+
+			usb3dr: usb@48300000 {
+				compatible = "st,stm32mp25-dwc3";
+				st,syscfg = <&syscfg 0x4800>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				ranges = <0x48300000 0x48300000 0x100000>;
+				access-controllers = <&rifsc 66>;
+				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
+				interrupts-extended = <&exti1 44 IRQ_TYPE_EDGE_RISING>;
+				status = "disabled";
+
+				dwc3: usb@48300000 {
+					compatible = "snps,dwc3";
+					reg = <0x48300000 0x100000>;
+					interrupts = <GIC_SPI 228 IRQ_TYPE_LEVEL_HIGH>;
+					clock-names = "ref", "bus_early", "suspend";
+					clocks = <&rcc CK_KER_USB2PHY2>, <&rcc CK_BUS_USB3DR>,
+						 <&rcc CK_KER_USB2PHY2>;
+					resets = <&rcc USB3DR_R>;
+					phys = <&usb2_phy2>;
+					phy-names = "usb2-phy";
+					wakeup-source;
+				};
+			};
+		};
+
+		risaf1: risaf@420a0000 {
+			compatible = "st,stm32mp25-risaf";
+			reg = <0x420a0000 0x1000>;
+			clocks = <&rcc CK_BUS_BKPSRAM>;
+			st,mem-map = <0x0 0x42000000 0x0 0x2000>;
+		};
+
+		risaf4: risaf@420d0000 {
+			compatible = "st,stm32mp25-risaf-enc";
+			reg = <0x420d0000 0x1000>;
+			clocks = <&rcc CK_BUS_RISAF4>;
+			st,mem-map = <0x0 0x80000000 0x1 0x00000000>;
 		};
 
 		bsec: efuse@44000000 {
@@ -2017,13 +2076,17 @@
 				reg = <0x24 0x4>;
 			};
 
+			vrefint: vrefin-cal@1b8 {
+				reg = <0x1b8 0x2>;
+			};
+
 			package_otp@1e8 {
 				reg = <0x1e8 0x1>;
 				bits = <0 3>;
 			};
 		};
 
-		dts: dts@44070000 {
+		dts: thermal-sensor@44070000 {
 			compatible = "moortec,mr75203";
 			reg = <0x44070000 0x80>,
 			      <0x44070080 0x180>,
@@ -2032,10 +2095,16 @@
 			reg-names = "common", "ts", "pd", "vm";
 			clocks = <&rcc CK_KER_DTS>;
 			resets = <&rcc DTS_R>;
-			power-domains = <&CLUSTER_PD>;
 			#thermal-sensor-cells = <1>;
 		};
 
+		hdp: pinctrl@44090000 {
+			compatible = "st,stm32mp-hdp";
+			reg = <0x44090000 0x400>;
+			clocks = <&rcc CK_BUS_HDP>;
+			status = "disabled";
+		};
+
 		rcc: clock-controller@44200000 {
 			compatible = "st,stm32mp25-rcc";
 			reg = <0x44200000 0x10000>;
@@ -2415,6 +2484,26 @@
 			};
 		};
 
+		pinctrl_z: pinctrl@46200000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			compatible = "st,stm32mp257-z-pinctrl";
+			ranges = <0 0x46200000 0x400>;
+			interrupt-parent = <&exti1>;
+
+			gpioz: gpio@46200000 {
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <2>;
+				reg = <0 0x400>;
+				clocks = <&scmi_clk CK_SCMI_GPIOZ>;
+				st,bank-name = "GPIOZ";
+				st,bank-ioport = <11>;
+				status = "disabled";
+			};
+		};
+
 		exti2: interrupt-controller@46230000 {
 			compatible = "st,stm32mp1-exti";
 			interrupt-controller;
@@ -2494,26 +2583,6 @@
 				<&intc GIC_SPI 213 IRQ_TYPE_LEVEL_HIGH>;	/* EXTI_70 */
 		};
 
-		pinctrl_z: pinctrl@46200000 {
-			#address-cells = <1>;
-			#size-cells = <1>;
-			compatible = "st,stm32mp257-z-pinctrl";
-			ranges = <0 0x46200000 0x400>;
-			interrupt-parent = <&exti1>;
-
-			gpioz: gpio@46200000 {
-				gpio-controller;
-				#gpio-cells = <2>;
-				interrupt-controller;
-				#interrupt-cells = <2>;
-				reg = <0 0x400>;
-				clocks = <&scmi_clk CK_SCMI_GPIOZ>;
-				st,bank-name = "GPIOZ";
-				st,bank-ioport = <11>;
-				status = "disabled";
-			};
-		};
-
 		fmc: memory-controller@48200000 {
 			compatible = "st,stm32mp25-fmc2-ebi";
 			reg = <0x48200000 0x400>;
@@ -2565,7 +2634,8 @@
 		compatible = "st,mlahb", "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <1>;
-		ranges = <0x0 0x0 0x0 0>;
+		ranges = <0x0 0x0 0x0 0xfffffffc>;
+		dma-ranges = <0x0 0x0 0x0 0xfffffffc>;
 
 		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
@@ -2582,4 +2652,5 @@
 			status = "disabled";
 		};
 	};
+
 };
--- a/arch/arm64/boot/dts/st/stm32mp233.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp233.dtsi
@@ -34,16 +34,17 @@
 	};
 
 	timer {
-		compatible = "arm,armv8-timer";
-		interrupt-parent = <&intc>;
 		interrupts = <GIC_PPI 13 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>,
 			     <GIC_PPI 14 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>,
 			     <GIC_PPI 11 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>,
 			     <GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>;
-		arm,no-tick-in-suspend;
 	};
 };
 
+&intc {
+	interrupts = <GIC_PPI 9 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>;
+};
+
 &optee {
 	interrupts = <GIC_PPI 15 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>;
 };
